app/controllers/concerns/spotlight/base.rb

```diff
@@ -43,6 +42,16 @@ def autocomplete_json_response_for_document(doc)
         iiif_manifest: doc[Spotlight::Engine.config.iiif_manifest_field]
       }
     end
-    # rubocop:enable Metrics/AbcSize
+
+    # Some pipeline steps may return an array instead of a string
```

Comment:
All pipeline steps are supposed to return an array except for the last join step. I believe it is the pipeline in DLME is not behaving.  I don't think we need to make this change in Spotlight.

Reply:
The join step in DLME is here https://github.com/sul-dlss/dlme/blob/main/app/processors/join.rb#L8 . It explicitly disallows the joining of the array into a string if it's a JSON API request (i.e. CatalogController request, JSON format). I added comments to the related DLME GitHub issue here: https://github.com/sul-dlss/dlme/issues/1705#issuecomment-1464717444 .  

If all pipeline steps are supposed to have a last join step that transforms the returned value into a string, is there another approach within the DLME Join we could take? I didn't want to override the original logic which seemed to prevent returned a join result if it's a JSON API request.

Reply:
I would consider having a separate pipeline for JSON display than for HTML display. 

If you didn't want to go that route, could you consider moving these changes from this PR into DLME instead? That way we don't have to add additional complexity in Spotlight that nobody else needs.

Reply:
I can follow up on slack with folks regarding the pipeline.  I'll close this PR for now.